unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/SceneObjectCatalogController.cs

```diff
@@ -0,0 +1,389 @@
+
+    public void ShowFavorites()
+    {
+        isFavoriteFilterActive = !isFavoriteFilterActive;
+
+        if (isFavoriteFilterActive)
+        {
+            string favoriteName = "Favorites";
+            List<Dictionary<string, List<SceneObject>>> favorites = new List<Dictionary<string, List<SceneObject>>>();
+            foreach (SceneAssetPack assetPack in CatalogController.sceneObjectCatalog.GetValues().ToList())
+            {
+                foreach (SceneObject sceneObject in assetPack.assets)
+                {
+                    foreach (SceneObject favObject in favoritesSceneObjects)
+                    {
+                        if (favObject != null)
+                        {
+                            if (sceneObject.id == favObject.id && sceneObject.asset_pack_id == favObject.asset_pack_id)
+                            {
+                                bool foundCategory = false;
+                                foreach (Dictionary<string, List<SceneObject>> groupedSceneObjects in favorites)
+                                {
+                                    if (groupedSceneObjects.ContainsKey(favoriteName))
+                                    {
+                                        foundCategory = true;
+                                        if (!groupedSceneObjects[favoriteName].Contains(sceneObject))
+                                            groupedSceneObjects[favoriteName].Add(sceneObject);
+                                    }
+                                }
+                                if (!foundCategory)
+                                {
+                                    Dictionary<string, List<SceneObject>> groupedSceneObjects = new Dictionary<string, List<SceneObject>>();
+                                    groupedSceneObjects.Add(favoriteName, new List<SceneObject>() { sceneObject });
+                                    favorites.Add(groupedSceneObjects);
+                                }
+                                break;
+
+                            }
+                        }
+                    }
+                }
+            }
+            catalogTitleTxt.text = favoriteName;
+            ShowCatalogContent();
+            catalogGroupListView.SetContent(favorites);
+        }
+        else
+        {
+            ShowAssetsPacks();
+        }
+       
+    }
+    public void AsignFavorite(SceneObject sceneObject, CatalogItemAdapter adapter)
+    {
+
+        if (!favoritesSceneObjects.Contains(sceneObject))
+        {
+            favoritesSceneObjects.Add(sceneObject);
+
+            int index = quickBarShortcutsSceneObjects.Count;
+            int cont = 0;
+            foreach (SceneObject sceneObjectIteration in quickBarShortcutsSceneObjects)
+            {
+                if (sceneObjectIteration == null)
+                {
+                    index = cont;
+                    break;
+                }
+                cont++;
+            }
```

Comment:
This should be a `private int FindEmptyShortcutSlot()` it also enforces my feeling that this shortcut thing should be a different controller/view.